databases_preview.go

```diff
@@ -0,0 +1,123 @@
+package scalingo
+
+import (
+	"context"
+	stderrors "errors"
+
+	"github.com/Scalingo/go-utils/errors/v2"
+)
+
+const databasesResource = "databases"
+
+var ErrDatabaseNotFound = stderrors.New("database not found")
+
+type PreviewAPI interface {
+	DatabasesPreviewService
+}
+
+type DatabasesPreviewService interface {
+	DatabaseCreate(ctx context.Context, params DatabaseCreateParams) (DatabaseNG, error)
+	DatabasesList(ctx context.Context) ([]DatabaseNG, error)
+	DatabaseShow(ctx context.Context, appID string) (DatabaseNG, error)
+	DatabaseDestroy(ctx context.Context, appID string) error
+}
+
+var _ DatabasesPreviewService = (*PreviewClient)(nil)
+
+type DatabaseCreateParams struct {
+	AddonProviderID string `json:"addon_provider_id"`
+	PlanID          string `json:"plan_id"`
+	DatabaseName    string `json:"database_name"`
+	ProjectID       string `json:"project_id,omitempty"`
+}
+
+// DatabaseNG stands for Database Next Generation.
+type DatabaseNG struct {
+	App      App       `json:"app"`
+	Addon    Addon     `json:"addon"`
+	Database *Database `json:"database,omitempty"`
+}
+
+type PreviewClient struct {
+	parent *Client
+}
+
+func NewPreviewClient(parent *Client) PreviewClient {
```

Comment:
New should return a pointer

```suggestion
func NewPreviewClient(parent *Client) *PreviewClient {
```